internal/transform/trade.go

```diff
@@ -0,0 +1,210 @@
+		}
+
+		var outputBaseAssetType, outputBaseAssetCode, outputBaseAssetIssuer string
+		err = claimOffer.AssetSold.Extract(&outputBaseAssetType, &outputBaseAssetCode, &outputBaseAssetIssuer)
+		if err != nil {
+			return []TradeOutput{}, err
+		}
+
+		outputBaseAmount := int64(claimOffer.AmountSold)
+		if outputBaseAmount < 0 {
+			return []TradeOutput{}, fmt.Errorf("Amount sold is negative (%d) for operation at index %d", outputBaseAmount, operationIndex)
+		}
+
+		outputCounterAccountAddress, err := utils.GetAccountAddressFromMuxedAccount(sourceAccount)
+		if err != nil {
+			return []TradeOutput{}, err
+		}
+
+		var outputCounterAssetType, outputCounterAssetCode, outputCounterAssetIssuer string
+		err = claimOffer.AssetBought.Extract(&outputCounterAssetType, &outputCounterAssetCode, &outputCounterAssetIssuer)
+		if err != nil {
+			return []TradeOutput{}, err
+		}
+
+		outputCounterAmount := int64(claimOffer.AmountBought)
+		if outputCounterAmount < 0 {
+			return []TradeOutput{}, fmt.Errorf("Amount bought is negative (%d) for operation at index %d", outputCounterAmount, operationIndex)
+		}
+
+		if outputBaseAmount == 0 && outputCounterAmount == 0 {
+			return []TradeOutput{}, fmt.Errorf("Both base and counter amount are 0 for operation at index %d", operationIndex)
+		}
+
+		// Final price should be buy / sell
+		outputPriceN, outputPriceD := outputCounterAmount, outputBaseAmount
+
+		if err != nil {
+			return []TradeOutput{}, err
+		}
+
+		outputBaseIsSeller := true
+
+		trade := TradeOutput{
+			Order:                 outputOrder,
+			LedgerClosedAt:        outputLedgerClosedAt,
+			OfferID:               outputOfferID,
+			BaseAccountAddress:    outputBaseAccountAddress,
+			BaseAssetType:         outputBaseAssetType,
+			BaseAssetCode:         outputBaseAssetCode,
+			BaseAssetIssuer:       outputBaseAssetIssuer,
+			BaseAmount:            outputBaseAmount,
+			CounterAccountAddress: outputCounterAccountAddress,
+			CounterAssetType:      outputCounterAssetType,
+			CounterAssetCode:      outputCounterAssetCode,
+			CounterAssetIssuer:    outputCounterAssetIssuer,
+			CounterAmount:         outputCounterAmount,
+			BaseIsSeller:          outputBaseIsSeller,
+			PriceN:                outputPriceN,
+			PriceD:                outputPriceD,
+		}
+
+		transformedTrades = append(transformedTrades, trade)
+	}
+	return transformedTrades, nil
+}
+
+func extractClaimedOffers(operationResults []xdr.OperationResult, operationIndex int32, operationType xdr.OperationType) (claimedOffers []xdr.ClaimOfferAtom, err error) {
+	if operationIndex >= int32(len(operationResults)) {
+		err = fmt.Errorf("Operation index of %d is out of bounds in result slice (len = %d)", operationIndex, len(operationResults))
+		return
+	}
+
+	if operationResults[operationIndex].Tr == nil {
+		err = fmt.Errorf("Could not get result Tr for operation at index %d", operationIndex)
+		return
+	}
+
+	operationTr, ok := operationResults[operationIndex].GetTr()
```

Comment:
Ok-checking can be shortened here (and elsewhere below) with the "comma ok" idiom in Go. You can find it on [this page](https://golang.org/doc/effective_go.html) by searching for "comma ok".